src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1889,6 +1910,10 @@ UniqueGenerator<int64_t> dateTimeGenerator(const Node& node, GeneratorArgs gener
          [](const Node& node, GeneratorArgs generatorArgs) {
              return std::make_unique<RandomDateGenerator>(node, generatorArgs);
          }},
+        {"^Date",
```

Comment:
While here, shoudl  IncDateGenerator be in this list also?

Reply:
it's already in the array on line 1509.

We should probably be better in the way we group the items in the list but I am going to leave this for now.

Reply:
That is different. That list is for all generators and does not assume a type. This is for the things that can return a date type. I'm trying to think of their uses -- it might not make sense to include it here, but I can't come up with a reason right now.

Reply:
:+1: this doesn't really make sense for {^Date} but it should have been in here (and there are maybe other cases that do make sense). I've added it and created a new test to exercise it.